### PR TITLE
Filter menu 

### LINKED ISSUE #16: Make filter menus to allow sorting of results
**Is your feature request related to a problem? Please describe.**
Add filter buttons to sort Pokemon by type, abilities, height, weight, alphabetical order, and Dex number.

**Describe the solution you'd like**
Use functions and hooks to sort pokemons.

### PR DESCRIPTION
**Pull request by connielion**
## Changes
1. Used hooks to determine if user searched, or clicked sort.
2. Filter & sort menu added. Sorting in order working.

## Purpose
Let user sort pokemons by name, dex number, type, ability, weight, and height.

## Approach
Used enumeration to determine sort type, hooks for state variables for pokemons array and user search boolean, and function for generating PokeCards and filter/sort. 
Closes #16

### CODE REVIEW

**Review by Gjhnsn — APPROVED** [review]
Great work!

File: src/components/Filter.js [hunk 1]
```diff
@@ -1,22 +1,27 @@
...
-  const [abilityOptions, setAbilityOptions] = useState([]);
+function Filter({
+  selectTypeOption,
+  setSelectTypeOption,
+  selectAbilityOptions,
+  setSelectAbilityOption,
+  abilityOptions,
+  setAbilityOptions,
+  setSortType
+}) {
```
Comment (BlackCubes) [inline src/components/Filter.js:14]: Nice destructure!

File: src/components/Filter.js [hunk 2]
```diff
@@ -38,7 +43,7 @@ function Filter() {
         };
         return choice;
       });
-      // sort array of objects, with keys: "value" & "label" alphabetically
+      // sort array of objects by "value" key alphabetically
```
Comment (BlackCubes) [inline src/components/Filter.js:46]: Nice grammar check!

File: src/components/Filter.js [hunk 3]
```diff
@@ -147,14 +151,13 @@ function Filter() {
...
         options={abilityOptions}
       />
-      <div>A-Z</div>
-      <div>Height</div>
-      <div>Weight</div>
-      <div>Dex Number</div>
+      <button onClick={()=>setSortType(SortTypes.ABC)}>A-Z</button>
+      <button onClick={()=>setSortType(SortTypes.HEIGHT)}>Height</button>
+      <button onClick={()=>setSortType(SortTypes.WEIGHT)}>Weight</button>
+      <button onClick={()=>setSortType(SortTypes.DEX_NO)}>Dex No.</button>
```
Comment (BlackCubes) [inline src/components/Filter.js:160]: Nice use of curry!

File: src/components/Filter.js [hunk 4]
```diff
@@ -138,7 +143,6 @@ function Filter() {
       <div>
```
Comment (dannysanchez559) [inline src/components/Filter.js:143]: consider adding className to div, so its easier to inspect in browser when we're testing out things :)

**Review by BlackCubes — APPROVED** [review]
Everything looks good! Great job!

File: src/components/PokeList.js [hunk 5]
```diff
@@ -11,53 +12,129 @@ const PokeList = ({
...
+      return heightA - heightB;
+    });
+  };
+  const sortByDexNumber = (pokemonList) => {
+    return pokemonList.sort((a, b) => {
+      const id1 = a["id"];
+      const id2 = b["id"];
+      return id1 - id2;
+    });
+  };
```
Comment (dannysanchez559) [inline src/components/PokeList.js:70]: should we stick all these sort functions in their own js file during the refactor stage?

**Review by dannysanchez559 — APPROVED** [review]
Looking good, just read the comments on the styling and refactoring and let me know what you think.

**Pull request merged**